{
  try {
    if (iterator != null) {
      iterator.close();
      iterator=null;
    }
  }
 catch (  Exception e) {
    getLogger().error("Error in closing iterator",e);
  }
}
